Documentation/build-configure.md

```diff
@@ -0,0 +1,137 @@
+# Build configuration
+
+## Overview
+
+This file describes all the parameters of the configure script and their possible uses.
+For a quick help of available parameters, run `./configure --help`.
+
+## Parameters for building stage1 flavors
+
+#### `--with-stage1-flavors`
+
+This parameter takes a comma-separated list of all the flavors that the build system should assemble.
+Depending on a default stage1 image setup, this list is by default either empty or set to `coreos,kvm` for, respectively, detailed setup and flavor setup.
+Note that specifying this parameter does not necessarily means that rkt will use them in the end.
+Available flavors are:
+
+- `coreos` - it takes systemd and bash from CoreOS PXE image; uses systemd-nspawn
+- `kvm` - it takes systemd, bash and other binaries from CoreOS PXE image; uses lkvm
+- `src` - it builds systemd, takes bash from host at build time; uses built systemd-nspawn
+- `host` - it takes systemd and bash from host at runtime; uses systemd-nspawn from host
+
+The `host` flavor is probably best suited flavor for distributions that have strict rules about software sources.
+
+#### `--with-stage1-flavors-version-override`
+
+This parameter takes a version number to become a version of all the built flavors of stage1 image.
+Normally, without this parameter, the images have the same version as rkt itself.
+This parameter may be useful for distributions that often provide patched versions of upstream software without changing major/minor/patch version number, but instead add a numeric suffix.
+An example usage could be passing `--with-stage1-flavors-version-override=0.12.0-2`, so the new image will have a version `0.12.0-2` instead of `0.12.0`.
+This parameter also affects default stage1 image version in flavor setup.
+
+## Parameters for setting up default stage1 image
+
+The parameters described below affects handling of default stage1 image by rkt.
```

Comment:
The parameters described below affect the handling of rkt's default stage1 image.